### DISCUSSION TITLE
Apache cannot start because listen tcp :443: bind: permission denied

### DISCUSSION THREAD

**Discussion by baalchina** [nextcloud/all-in-one#1267] [Questions] [answered]
Hi, I am trying nextcloud aio in a centos 7 system(with selinux disabled), while I found my apache can not start(and any other container running successful).
Here is my log:
`
.....
{"level":"info","ts":1665835991.7811992,"logger":"tls","msg":"cleaning storage unit","description":"FileStorage:/mnt/data/caddy"}
{"level":"info","ts":1665835991.7812896,"logger":"tls","msg":"finished cleaning storage units"}
{"level":"info","ts":1665835991.7818909,"logger":"tls.cache.maintenance","msg":"stopped background certificate maintenance","cache":"0xc000339260"}
Error: loading initial config: loading new config: http app module: start: tcp: listening on :443: listen tcp :443: bind: permission denied
....
`

well, seems someone had port 443 bind or no permission.

In my centos host, I got this when i run netstat
```
tcp        0      0 0.0.0.0:443             0.0.0.0:*               LISTEN      28865/docker-proxy  
tcp        0      0 0.0.0.0:8443            0.0.0.0:*               LISTEN      25600/docker-proxy  
tcp6       0      0 :::443                  :::*                    LISTEN      28871/docker-proxy  
tcp6       0      0 :::8443                 :::*                    LISTEN      25607/docker-proxy
```

and after `docker stop $(docker ps -aq)`, there is no 443 port list.

Here is my docker command, so, where is the problem? Thanks!

```
sudo docker run \
--sig-proxy=false \
--name nextcloud-aio-mastercontainer \
--restart always \
--publish 80:80 \
--publish 8080:8080 \
--publish 8443:8443 \
-e SKIP_DOMAIN_VALIDATION=true \
-e NEXTCLOUD_DATADIR="/ncdata" \
--volume nextcloud_aio_mastercontainer:/ncdata/docker-aio-config \
--volume /var/run/docker.sock:/var/run/docker.sock:ro \
nextcloud/all-in-one:latest
```

Thank you.

================

Well, I tried some docker command

```
[root@nc-aio ~]# docker ps
CONTAINER ID   IMAGE                             COMMAND                  CREATED          STATUS                     PORTS                                                                                                                     NAMES
1a9961e042e0   nextcloud/aio-apache:latest       "start.sh /usr/bin/s…"   8 minutes ago    Up 8 minutes (healthy)     0.0.0.0:443->443/tcp, :::443->443/tcp                                                                                     nextcloud-aio-apache
```

seems nextcloud/aio-apache:latest had already had 443 in use?

**Comment by szaimen** [comment]
Which Docker version do you have installed? 
I just found this after googling which should be present in docker v20.10 and higher:
https://github.com/moby/moby/pull/41030

Also, is this docker rootless or normal docker?

**Comment by szaimen** [comment]
Can you also check if docker is correctly installed by following https://docs.docker.com/engine/install/centos/?

**Comment by baalchina** [comment]
Thank you szaimen!

I am running docker in root user, so it is normal docker?
And docker version is  20.10.19, build d85ef84, I installed docker using `curl -fsSL get.docker.com | sudo sh` which showed in the aio document.

**Comment by szaimen** [comment, marked as answer]
I found the problem: https://github.com/docker-library/haproxy/issues/159#issuecomment-853242578

You need centos8 or higher (kernel 4.11 or higher) in order to make it work.

**Reply by baalchina** [reply]
After changed to CentOS8, it works! Thankyou!
